### PR TITLE
GPII-3999:  Apply the istio-proxy-manager solution to gpii-dataloader

### PR DESCRIPTION
**Pull request by klown**
@mrtyler @stepanstipl @amatas 

This pull request implements the same `istio-proxy-manager` as the `gpii-flushtokens` cronjob, but here for the `gpii-dataloader` job.  It is based on PR #478 and includes the changes from there.

I could have added it to the code in PR #478, but didn't want to disturb the status of that PR, namely, reviewed and ready to merge.  This PR is ready for review.

**NOTE**: As of 09-Aug-2019, this pull request references a tagged version of the [gpii-universal docker image](https://ci.gpii.net/job/docker-gpii-universal-master-calculate-tag/33/console):
```
repository: gpii/universal
tag: 20190809170605-f8485e9
```
**NOTE2**: Merging of this code depends on first merging the code in PR #478.

JIRA:  https://issues.gpii.net/browse/GPII-3999

cc:  @cindyli

<del>
**NOTE**: As of 09-Aug-2019, this pull request references @cindyli's development version of the docker image of gpii-universal:
```
repository: cindyqili/universal
tag: latest
```

This MUST be replaced with the official tagged docker image provided by CI's tag calculation process (https://ci.gpii.net/job/docker-gpii-universal-master-calculate-tag/...).
</del>

<del>
**NOTE**: as of 01-Aug-2019, this pull request uses the same tagged version of the gpii-universal docker image used by #438, specifically:
<pre>
repository: gpii/universal
tag: 20190801163411-26be63f
</pre>

**NOTE**:  as of 05-Jul-2019, this pull request depends on the tagged version of the gpii-universal docker image used by #438, specifically :
<pre>
repository: gpii/universal
tag: 20190627203047-1262ac8
</pre>
</del>

### CODE REVIEW

**Comment by stepanstipl** [comment]
Hi @klown, thanks, looks good to me 👍  (I've reviewed just the dataloader bit, but I assume we plan to merge this only after #383 just to make sure nothing else changed)

**Comment by klown** [comment]
Thanks @stepanstipl.  Works for me!

**Comment by klown** [comment]
@amatas wrote:
> LGTM, perhaps we need to add a note to the description about the dependency of #383 to be merged first.

Done.

File: shared/charts/gpii-dataloader/templates/job.yaml [hunk 1]
```diff
@@ -9,11 +9,33 @@ spec:
...
+            done;
+            echo 'deleteAndLoadSnapsets.sh started.';
+            while ps -o user | grep node > /dev/null;
+            do
+              echo 'Waiting for deleteAndLoadSnapsets.sh to finish...';
+              sleep 2;
+            done;
+            echo 'deleteAndLoadSnapsets.sh finished.';
+            echo 'Sending TERM to pilot-agent';
+            pkill pilot-agent;
```
Comment (mrtyler) [inline shared/charts/gpii-dataloader/templates/job.yaml]: Same question as here: https://github.com/gpii-ops/gpii-infra/pull/383#discussion_r309374031

Comment (klown) [reply shared/charts/gpii-dataloader/templates/job.yaml]: Same response:  added the loop.

**Comment by klown** [comment]
@mrtyler wrote:

> FYI Github is reporting a conflict with versions.yml, but I believe you have a plan for that.

With the latest merge of master and flushtokens, the conflict has vanished.

**Comment by mrtyler** [comment]
LGTM

**Comment by stepanstipl** [comment]
@klown & others - I did change base of this PR to some random one & back, in order to force GH to refresh diff of this PR (as the master changed in the meantime). It now shows only the desired - small - diff of actual changes. And it looks good.

**Comment by stepanstipl** [comment]
I've just bumped the dataloader Helm chart version & I'm merging this.

**Comment by stepanstipl** [comment]
done, deployed all the way to prod & all looks good, I’ve checked the logs and dataloader did run, did some work & then terminated nicely with istio. 👍

**Pull request merged**